src/game_launcher.cpp

```diff
@@ -240,14 +241,14 @@ game_launcher::game_launcher(const commandline_options& cmdline_opts, const char
 			else
 				multiplayer_server_ = "";
 		}
-	}
-	if (cmdline_opts_.username) {
-		preferences::disable_preferences_save();
-		preferences::set_login(*cmdline_opts_.username);
-	}
-	if (cmdline_opts_.password) {
-		preferences::disable_preferences_save();
-		preferences::set_password(*cmdline_opts_.password);
+		if (cmdline_opts_.username) {
+			preferences::disable_preferences_save();
+			preferences::set_login(*cmdline_opts_.username);
+			if (cmdline_opts_.password) {
+				preferences::disable_preferences_save();
```

Comment:
Based on preferences.cpp line 116 and the fact that write_preferences writes both files, I think this should work correctly as is.